File: src/com/practice/java/common/ThreadTest.java
package com.practice.java.common;

public class ThreadTest {

    public static void main(String[] args) {

        OddNumber oddNumber = new ThreadTest().new OddNumber();
        Thread tOdd = new Thread(oddNumber);
        tOdd.start();

        EvenNumber evenNumber = new ThreadTest().new EvenNumber();
        Thread tEven = new Thread(evenNumber);
        tEven.start();
    }

    private class OddNumber implements Runnable {
        int oddNumb = 1;

        @Override
        public void run() {
            try {
                while (oddNumb < 200) {
                    System.out.println(oddNumb);
                    oddNumb += 2;
                    Thread.sleep(50);
                }
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }

    private class EvenNumber implements Runnable {
        int evenNumb = 2;

        @Override
        public void run() {
            try {
                while (evenNumb <= 200) {
                    System.out.println(evenNumb);
                    evenNumb += 2;
                    Thread.sleep(50);
                }
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}

File: src/com/practice/kotlin/common/ThreadTest.kt
package com.practice.kotlin.common

class ThreadTest {

    private inner class OddNumber : Runnable {
        internal var oddNumb = 1

        override fun run() {
            try {
                while (oddNumb < 200) {
                    println(oddNumb)
                    oddNumb += 2
                    Thread.sleep(50)
                }
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

        }
    }

    private inner class EvenNumber : Runnable {
        internal var evenNumb = 2
        override fun run() {
            try {
                while (evenNumb <= 200) {
                    println(evenNumb)
                    evenNumb += 2
                    Thread.sleep(50)
                }
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {

            val oddNumber = ThreadTest().OddNumber()
            val tOdd = Thread(oddNumber)
            tOdd.start()

            val evenNumber = ThreadTest().EvenNumber()
            val tEven = Thread(evenNumber)
            tEven.start()
        }
    }
}
